[nmstate-2.1] ifaces: Ignore interface even mentioned in port list

cathay4t opened the pull request:
In old Python API, we ignore interfaces only depend on `interfaces`
section of desire state regardless it is mentioned in port list or not.

To be consistent, we changed the rust code to align with it.

Integration test case included.

cathay4t committed:
ifaces: Ignore interface even mentioned in port list

In old Python API, we ignore interfaces only depend on `interfaces`
section of desire state regardless it is mentioned in port list or not.

To be consistent, we changed the rust code to align with it.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera commented on rust/src/lib/ifaces/inter_ifaces_controller.rs:
```diff
@@ -213,38 +207,6 @@ fn handle_changed_ports_of_iface(
     Ok(())
 }
 
-// When desire desire a port which is ignored in current, we should
-// include this port also even it is already assigned to desired controller,
-// so that it could change state from ignore to up.
-fn include_ignored_iface_if_desired_in_port(
-    des_iface: &Interface,
-    cur_ifaces: &Interfaces,
-    pending_changes: &mut HashMap<
-        String,
-        (Option<String>, Option<InterfaceType>),
-    >,
-) {
-    if let Some(ports) = des_iface.ports().or_else(|| {
-        cur_ifaces
-            .get_iface(des_iface.name(), des_iface.iface_type())
-            .and_then(|i| i.ports())
-    }) {
-        for port_name in ports {
-            if let Some(cur_iface) = cur_ifaces.kernel_ifaces.get(port_name) {
-                if cur_iface.is_ignore() {
-                    pending_changes.insert(
-                        port_name.to_string(),
-                        (
-                            Some(des_iface.name().to_string()),
-                            Some(des_iface.iface_type()),
-                        ),
-                    );
-                }
-            }
-        }
-    }
-}
-
```
I don't think we should drop this function. Instead I would suggest to check that the bridge interface is UP in the desire state first, if not, skip the procedure... Without this function the `test_linux_bridge_partial_ignored()` unit test will fail and I think it is a valid use case.

What do you think?

cathay4t replied:
I have included a patch to fix that use case delicately.

cathay4t committed:
ifaces: Allow partial consuming port interface from ignore to up

Given a bridge `br0` with `eth1` and `eth2` ports, all of them are
`state: ignore`. When applying this state,

```yaml
interfaces:
- name: br0
  type: linux-bridge
  state: up
- name: eth1
  type: ethernet
  state: up
```

nmstate should partially convert br0 and eth1 to managed state and
leaving eth2 untouched.

Both unit and integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
